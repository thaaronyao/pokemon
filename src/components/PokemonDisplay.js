import React from 'react';

const PokemonDisplay = ({pokemon, getPokemonNum}) => {

    return (
        <div className="pokemonDisplay">
            <p>{pokemon.name}</p>
            <div className="placement">
                <img className="thumbnail" src={pokemon.image}/>
                <br/>
                <button onClick={() => getPokemonNum(pokemon.number)}>More Information</button>
            </div>
        </div>
    )
}

export default PokemonDisplay;