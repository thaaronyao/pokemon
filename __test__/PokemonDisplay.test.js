import React from 'react';
import renderer from 'react-test-renderer';

import PokemonDisplay from '../src/components/PokemonDisplay';

test('render test', () => {
    const props = {   
            image: "https://img.pokemondb.net/artwork/bulbasaur.jpg",
            name: "Bulbasaur"
        }
    const component = renderer.create(<PokemonDisplay pokemon={props}/>);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})
