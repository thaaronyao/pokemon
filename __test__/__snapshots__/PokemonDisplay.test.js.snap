// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test 1`] = `
<div
  className="pokemonDisplay"
>
  <p>
    Bulbasaur
  </p>
  <div
    className="placement"
  >
    <img
      className="thumbnail"
      src="https://img.pokemondb.net/artwork/bulbasaur.jpg"
    />
    <br />
    <button
      onClick={[Function]}
    >
      More Information
    </button>
  </div>
</div>
`;
